### PR TITLE
Terraform for vpc, subnets and network infrastructure

### PR DESCRIPTION
**Pull request by kaushikchandrashekar**

### CODE REVIEW

File: terraform/state-bucket/prod/main.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  region  = "ap-south-1"
+  profile = "ecs-workshop"
+  version = "= 1.43.0"
+}
+
+resource "aws_s3_bucket" "terraform_state_storage" {
+  bucket = "prod-ecs-workshop-terraform-state"
+  acl    = "private"
+  region = "ap-south-1"
+
```
Comment (prashantkalkar) [inline terraform/state-bucket/prod/main.tf:15]:
```suggestion
versioning {
    enabled = true
  }
```

File: terraform/state-bucket/prod/main.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  tags {
+    Name        = "prod-ecs-workshop-terraform-state"
+    Environment = "prod"
+  }
+
+  lifecycle {
+    prevent_destroy = false
```
Comment (prashantkalkar) [inline terraform/state-bucket/prod/main.tf]: Any reason this is mentioned. Isn't prevent_destroy false default?

File: terraform-modules/routes/main.tf [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  tags {
+    Name        = "${var.name}-private-route-table"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_route" "main_route" {
+  route_table_id         = "${var.main_route_table_id}"
+  destination_cidr_block = "${var.destination_cidr_block}"
+  gateway_id             = "${aws_internet_gateway.internet_gateway.id}"
```
Comment (prashantkalkar) [inline terraform-modules/routes/main.tf:43]: I don't think it is recommended practice to update the vpc main route table to allow traffic to the internet. Any Subnets created under the VPCs inherit the routes from the main routing table implicitly unless you associate a separate table to them. So it better to create a separate public table and explicitly assign it to public subnets. 
At least this is the case if done from aws console.

**Review by prashantkalkar — COMMENTED** [review]
Should we also add support from IPv6 traffic?

**Comment by prashantkalkar** [comment]
Is everyone going to create things inside the same vpc?

**Comment by prashantkalkar** [comment]
Do we create on more for DEV environment?

File: terraform/state-bucket/prod/main.tf (hunk 2 above)
Comment (kaushikchandrashekar) [reply terraform/state-bucket/prod/main.tf]: Intended for it to be true. Committed the fix

**Pull request merged**